ibis/backends/impala/__init__.py

```diff
@@ -137,6 +139,12 @@ def register_options(self):
         )
 
     def set_database(self, name):
+        warnings.warn(
```

Comment:
Random thought to be done later: should we consider using https://deprecation.readthedocs.io/en/latest to automate the annoyances of deprecation?

Reply:
Didn't know about the library, seems like a great idea.